api/envoy/config/trace/v2/trace.proto

```diff
@@ -198,6 +200,19 @@ message OpenCensusConfig {
   repeated TraceContext outgoing_trace_context = 9;
 }
 
+// Configuration for AWS X-Ray tracer.
+message XRayConfig {
+  // The endpoint of the X-Ray Daemon where the spans will be sent. Since by default daemon
+  // listens to localhost:2000, so the default value is 127.0.0.1:2000.
+  string daemon_endpoint = 1 [(validate.rules).string = {min_bytes: 1}];
+
+  // The custom name to name a X-Ray segment. By default will use cluster name.
+  string segment_name = 2;
+
+  // The location of custom sampling rule json file.
+  string json_file = 3;
```

Comment:
Is there a reason this is a file and not an inline string? Could it be a `DataSource`?

Reply:
>Is there a reason this is a file and not an inline string? 

Familiarity with how X-Ray already works in all the other SDKs.

Reply:
Would making it a `DataSource` provide user's a potential ability to avoid storing things on the filesystem? I'd just like to ensure we don't force an unnecessary restriction on operators here.

Reply:
I've already made the change to DataSource